client/my-sites/email/form/mailboxes/components/field.tsx

```diff
@@ -0,0 +1,116 @@
+import { FormInputValidation } from '@automattic/components';
+import { useState } from 'react';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormPasswordInput from 'calypso/components/forms/form-password-input';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import FormTextInputWithAffixes from 'calypso/components/forms/form-text-input-with-affixes';
+import { useGetDefaultFieldLabelText } from 'calypso/my-sites/email/form/mailboxes/components/use-get-default-field-label-text';
+import type {
+	MailboxFormFieldBase,
+	MutableFormFieldNames,
+} from 'calypso/my-sites/email/form/mailboxes/types';
+import type { ChangeEvent } from 'react';
+
+interface MailboxFormFieldProps {
+	domains?: string[];
+	field: MailboxFormFieldBase< string >;
+	isPasswordField?: boolean;
+	lowerCaseChangeValue?: boolean;
+	notifyFieldValueChanged?: ( field: MailboxFormFieldBase< string > ) => void;
+	requestFieldValidation: ( field: MailboxFormFieldBase< string > ) => void;
```

Comment:
Feels more natural to me to name it as:
```suggestion
	onFieldValueChanged?: ( field: MailboxFormFieldBase< string > ) => void;
	onRequestFieldValidation: ( field: MailboxFormFieldBase< string > ) => void;
```
```suggestion
	notifyFieldValueChanged?: ( field: MailboxFormFieldBase< string > ) => void;
	requestFieldValidation: ( field: MailboxFormFieldBase< string > ) => void;
```